client/v3/logger.go

```diff
@@ -51,7 +51,7 @@ func etcdClientDebugLevel() zapcore.Level {
 		return zapcore.InfoLevel
 	}
 	var l zapcore.Level
-	if err := l.Set(envLevel); err == nil {
+	if err := l.Set(envLevel); err != nil {
```

Comment:
The change should be good. But the message below is a little confusing. Is `ETCD_CLIENT_DEBUG` really deprecated? cc @serathius 
```
Deprecated env ETCD_CLIENT_DEBUG value. Using default level: 'info'
```

Reply:
I parsed that sentence as the _value_ being deprecated, rather than the setting.

https://github.com/etcd-io/etcd/pull/12786 introduced this code and explicitly brought back ETCD_CLIENT_DEBUG.

Reply:
Doesn't the following message  make more sense? @ptabor 
```
Invalid value for 'ETCD_CLIENT_DEBUG'. Using default level: 'info'
```

Reply:
Yes. That would be definitely better.